.github/GSOC.md

```diff
@@ -0,0 +1,124 @@
+# GSOC Participant Guide
+
+![](https://raw.githubusercontent.com/checkstyle/resources/master/img/checkstyle-logos/checkstyle-logo-260x99.png)
+
+---------------------------------
+
+**This guide serves to set clear expectations and provide guidance for Google Summer of Code (GSoC)
+participants.**
+
+:sun_with_face: Thanks for your interest in participating in GSoC with Checkstyle! :sun_with_face:
+
+## Contents
+ - [Getting Started](#rocket-getting-started)
+ - [Additional Tips](#star-additional-tips)
+ - [Your Proposal](#technologist-your-proposal)
+
+## :rocket: Getting Started
+
+Thanks for your interest in participating in GSoC with Checkstyle! Joining Checkstyle for the summer
+is a great way to learn about open source software development and contribute to a widely-used
+project. We are excited to work with you and help you grow as a software developer. You don't
+need to be an expert in Java or Checkstyle to participate. We are looking for participants who are
+eager to learn, work hard, and are excited to contribute to open source software. Additionally, we
+want you to stick around after GSoC and become a long-term member of our community :smile:.
+
+Please take the time to read through this guide and familiarize yourself with the project. If you
+have any questions, please don't hesitate to ask. We are here to help you succeed. If you have
+not already done so, please read through the following documentation provided by Google:
+
+ - [GSoC Participant Guide](https://summerofcode.withgoogle.com/rules/).
+ - [GSoC FAQ](https://developers.google.com/open-source/gsoc/faq).
+
+### Recommended Steps
+
+1. **Get to know the project.** Spend some time reading through the Checkstyle documentation and
+   familiarizing yourself with the project. You can find the documentation at
+   [https://checkstyle.org/](https://checkstyle.org/).
+2. **Join the Checkstyle community.**
+   1. Join the Checkstyle group on [Gitter](https://gitter.im/checkstyle/checkstyle).
+      This is a great place to ask questions and get to know the community.
+   2. Join the Checkstyle group on [Google Groups](https://groups.google.com/forum/#!forum/checkstyle).
+      This is another place to ask questions and get to know the community.
```

Comment:
Should we also add Discord Server link? As it has separate channels for each projects and it'll help users to ask doubts in more easier way. Plus it also has previously discussed topics so it'll be a good benefit for users.

Reply:
I updated README to have a link to general discord.

GSOC.md is not finished, but I will migrate GSOC.md to discord too.
project specific links should be hidden to let students focus on contributions first and we will invite only people who did at least 1-2 PRs.

Reply:
Hmm... I guess invite only rule will be difficult to manage as there'll be lot of contributors and we'll need to check contributions of all of them. We can public the project channels when GSoC officially announces project lists of all orgs but we'll ask users to share their draft proposal link in DMs so other users cannot see them or copy from them.

Reply:
> We can public the project channels when GSoC officially announces project lists of all orgs

I assume you haven't realized none of the projects in discord have been made public. They are all still private. The GH projects were also private, but looks like they may have been made public on their closure.

Reply:
By project channels, I meant the normal channels which were created for each projects which normals users can see, not the private channel that we create for user when he gets selected into gsoc, I have attached the publicly available channels screenshot.

![IMG-20241027-WA0004.jpg](https://github.com/user-attachments/assets/46c5714e-dd2f-4060-9c75-acd64e36c19e)



We can keep these normal project channels private until gsoc announces project list and if we decide give its access to only users who have atleast 2 PRs merged then we'll need to cross check it for all users and it'll require extra time.

Reply:
Sorry, we were talking about GSoC so  I thought you meant the GSoC channels.